bokeh/client/states.py

```diff
@@ -50,39 +58,45 @@ class ErrorReason(Enum):
     HTTP_ERROR      = auto()
     NETWORK_ERROR   = auto()
 
-class NOT_YET_CONNECTED:
+class State(metaclass=ABCMeta):
```

Comment:
Perhaps this would be better now as a `Protocol`?

Reply:
Not really. Actually both approaches don't give any static feedback whether sub-classes are implemented correctly or not (no wonder as there's no `@abstract` decorator; ours does nothing). At least with `ABC` we will get a meaningful exception at runtime.